crates/iota-core/src/authority/authority_per_epoch_store.rs

```diff
@@ -3934,10 +3934,28 @@ impl AuthorityPerEpochStore {
                 panic!("process_consensus_transaction called with end-of-publish transaction");
             }
             SequencedConsensusTransactionKind::External(ConsensusTransaction {
-                kind: ConsensusTransactionKind::MisbehaviorReport(_authority, _report, _),
+                kind: ConsensusTransactionKind::MisbehaviorReport(authority, report, _),
                 ..
             }) => {
-                // Validate report
+                let authority_index = self
+                    .committee
+                    .authority_index(authority)
+                    .expect("authority in committee");
+                // Check validity of the report
+                if !report.verify(self.committee.num_members()) {
+                    // Since this verification happens after consensus, all authorities already
+                    // agreed on the set of messages that would be verified by this method. Then, we
+                    // can update scores according to the validation result directly, without the
```

Comment:
This comment is a bit long-winded but I think the message is a bit more simple: We have consensus on the inclusion of this misbehavior report in the DAG. Now we just objectively verify and either count it as valid or not.